api/src/main/java/uk/gov/hmcts/payment/api/dto/mapper/PaymentDtoMapper.java

```diff
@@ -99,30 +115,46 @@ public PaymentDto toBulkScanPaymentStrategicDto(Payment payment, String paymentG
             .build();
     }
 
-
-
-
-
-    public PaymentDto toPciPalCardPaymentDto(PaymentFeeLink paymentFeeLink, String link) {
-        Payment payment = paymentFeeLink.getPayments().get(0);
+    public PaymentDto toPciPalCardPaymentDto(PaymentFeeLink paymentFeeLink, Payment payment, String link) {
         return PaymentDto.payment2DtoWith()
             .status(PayStatusToPayHubStatus.valueOf(payment.getStatus().toLowerCase()).getMappedStatus())
             .reference(payment.getReference())
             .paymentGroupReference(paymentFeeLink.getPaymentReference())
-            .fees(paymentFeeLink.getFees() != null ? toFeeDtos(paymentFeeLink.getFees()) : null)
+            .fees(toFeeDtos(paymentFeeLink.getFees()))
```

Comment:
Hi @Jennings-HMCTS - Are we handling this condition?

Reply:
function toPciPalCardPaymentDto with params paymentfeelink and link is renamed to toCreateCardPaymentTelephonyResponse at line 129 where you can see those conditions.

Reply:
Got it. Approved.